### ISSUE TITLE
Text that is not empty on the next line of a hyperlink in a rich text widget cannot be wrapped correctly

### ISSUE THREAD

**Issue by qiwentaidi** [fyne-io/fyne#4340]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

1、Text that is not empty on the next line of a hyperlink in a rich text widget cannot be wrapped correctly
2、Unable to recognize HTML code

### How to reproduce

use example code

### Screenshots

![image](https://github.com/fyne-io/fyne/assets/104491549/e9de2c4b-9fc7-4fd6-bfae-73a13d580d54)

### Example code

```go 
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

const info = `<h1 align="center">test title</h1>
<p align="center">hello<br/><br/></p>

> 123456789
111111
1111

[google](https://google.com/)





can't wrap
`

func main() {
	a := app.New()
	w := a.NewWindow("Test RichText")
	rt := widget.NewRichTextFromMarkdown(info)
	w.SetContent(rt)
	w.ShowAndRun()
}

````

### Fyne version

fyne.io/fyne/v2 v2.4.1-rc1.0.20231020171735-2921c115f4cc

### Go compiler version

go1.20.1 windows/amd64

### Operating system and version

windows11

**Comment by dweymouth** [comment]
Probably related to #3393, it's possible they even share the same root cause. I wouldn't consider this a duplicate though, it's still its own issue.

**Comment by andydotxyz** [comment]
I actually think they are separate issues and the use of "wrap" here is in a slightly different context.
Notice that wrapping is not turned on in the code snippet. The problem seems to be that without some other content after an href it seems to get confused and does not respect the newline.

Just for clarity, if I turn on wrapping you can see that the wrapping applies to the text (and not the hyperlink as per #3393) but that the first letter of the next word is stuck to the link.

![hl-wrap](https://github.com/fyne-io/fyne/assets/294436/282a5a14-9ac7-421c-85ab-de08adb65885)

**Comment by andydotxyz** [comment]
(note also the HTML included is not required to replicate, and indeed not supported).

**Comment by ErikKalkoken** [comment]
I think I found a workaround for this issue.

The issue occurs when a link is at the end of a line. In the below example `Fyne` will be rendered on the same line as the link.

```text
"[google](http://www.google.com)\n\nFyne"
```

As a workaround an invisible space can be added at the end of the line with the URL. Then it be correctly wrapped and `Fyne` appears on a next line.

```text
"[google](http://www.google.com) \n\nFyne"
```

Note that a normal space will not work, because those are ignored by the markdown parser.
Instead you can use an [Unicode EM space (U+2003)](https://www.compart.com/en/unicode/U+2003)

**Comment by andydotxyz** [comment]
This is resolved on develop ready for v2.5.0
